File: prextras-agent/src/main/java/com/nicologies/prextras/BuildServiceFactory.java
package com.nicologies.prextras;

import com.nicologies.prextras.common.PrExtrasConstants;
import jetbrains.buildServer.agent.AgentBuildRunnerInfo;
import jetbrains.buildServer.agent.BuildAgentConfiguration;
import jetbrains.buildServer.agent.runner.CommandLineBuildService;
import jetbrains.buildServer.agent.runner.CommandLineBuildServiceFactory;
import org.apache.log4j.Logger;
import org.jetbrains.annotations.NotNull;

public class BuildServiceFactory implements CommandLineBuildServiceFactory, AgentBuildRunnerInfo {
    private static final Logger LOG = Logger.getLogger(BuildServiceFactory.class);

    public BuildServiceFactory() {
    }

    @NotNull
    public String getType() {
        return PrExtrasConstants.RunnerType;
    }

    public boolean canRun(@NotNull final BuildAgentConfiguration agentConfiguration) {
        return true;
    }


    @NotNull
    public CommandLineBuildService createService() {
        return new BuildService();
    }

    @NotNull
    public AgentBuildRunnerInfo getBuildRunnerInfo() {
        return this;
    }
}

File: prextras-agent/src/main/kotlin/com/nicologies/prextras/BuildServiceFactory.kt
package com.nicologies.prextras

import com.nicologies.prextras.common.PrExtrasConstants
import jetbrains.buildServer.agent.AgentBuildRunnerInfo
import jetbrains.buildServer.agent.BuildAgentConfiguration
import jetbrains.buildServer.agent.runner.CommandLineBuildService
import jetbrains.buildServer.agent.runner.CommandLineBuildServiceFactory

class BuildServiceFactory : CommandLineBuildServiceFactory, AgentBuildRunnerInfo {

    override fun getType(): String {
        return PrExtrasConstants.RunnerType
    }

    override fun canRun(agentConfiguration: BuildAgentConfiguration): Boolean {
        return true
    }


    override fun createService(): CommandLineBuildService {
        return BuildService()
    }

    override fun getBuildRunnerInfo(): AgentBuildRunnerInfo {
        return this
    }
}